### ISSUE TITLE
Creating new records not working with `BelongsTo` relationship 

### ISSUE THREAD

**Issue by MaximeWillinger** [laravel-filament/filament#2576]
### Package

filament/forms

### Package Version

v2.12.10

### Laravel Version

v9.13.0

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1.3

### Bug description

When using a `BelongsToSelect` with a `createOptionForm()`, the `+` button to open the creation form does nothing.

### Steps to reproduce

Create a Relation Manager and add a `BelongsToSelect` form field as following:

```php
Forms\Components\BelongsToSelect::make('user')->relationship('user', 'id')->createOptionForm([
    Forms\Components\TextInput::make('id')->required(),
]),
```

Then open the modal and click on the `+`. Nothing happens.

**Comment by danharrin** [comment]
Fixed by #2599

**Comment by MaximeWillinger** [comment]
Thanks @danharrin 🥳
